mod/node-core/pkg/builder/baseapp_options.go

```diff
@@ -98,3 +84,28 @@ func DefaultServiceOptions[
 		cometbft.SetChainID[LoggerT](chainID),
 	}
 }
+
+func loadChainIDFromGenesis(appOpts config.AppOptions) (string, error) {
+	var (
+		homeDir = cast.ToString(appOpts.Get(flags.FlagHome))
+		fp      = filepath.Join(homeDir, "config", "genesis.json")
+	)
+
+	f, err := os.Open(filepath.Clean(fp))
```

Comment:
Found [here](https://securego.io/docs/rules/g304.html) the suggestion to fix it up